src/System.Net.WebHeaderCollection/src/System/Net/WebHeaderCollection.cs

```diff
@@ -16,19 +19,38 @@ internal enum WebHeaderCollectionType : byte
         WebResponse,
     }
 
-    public sealed class WebHeaderCollection : IEnumerable
+    [Serializable]
+    public class WebHeaderCollection : NameValueCollection, ISerializable
```

Comment:
Note that #12896 added the serialization implementation for WebHeaderCollection.  It looks basically identical to what you've done here.